grammar.js

```diff
@@ -215,17 +219,24 @@ module.exports = grammar({
           $._namespace_given_by_type,
           $.namespace_wildcard,
           $.identifier,
-          $.renamed_identifier
+          $.arrow_renamed_identifier,
+          $.as_renamed_identifier
         )),
       '}'
     ),
 
     // deprecated: Remove when highlight query is updated for Neovim
     _import_selectors: $ => alias($.namespace_selectors, $.import_selectors),
 
-    renamed_identifier: $ => seq(
+    arrow_renamed_identifier: $ => seq(
```

Comment:
I split the renamed_identifier node as it was a low-tech solution to prevent parsing code that is invalid in Scala 3.

If that's unacceptable, then we can loosen the restriction and use the old renamed_identifier node and just parse incorrect code, if it's for the purspose of highlighting anyways.

Reply:
My opinion would be that this is probably fine.